fahrschein/src/main/java/org/zalando/fahrschein/ManagedCursorManager.java

```diff
@@ -73,7 +74,15 @@ void setStreamId(String streamId) {
     private final ObjectMapper objectMapper;
     private final Map<String, SubscriptionStream> streams;
 
+    public ManagedCursorManager(URI baseUri, ClientHttpRequestFactory clientHttpRequestFactory, AccessTokenProvider accessTokenProvider) {
+        this(baseUri, wrapClientHttpRequestFactory(clientHttpRequestFactory, accessTokenProvider), true);
+    }
+
     public ManagedCursorManager(URI baseUri, ClientHttpRequestFactory clientHttpRequestFactory) {
+        this(baseUri, wrapClientHttpRequestFactory(clientHttpRequestFactory, null), true);
+    }
+
+    ManagedCursorManager(URI baseUri, ClientHttpRequestFactory clientHttpRequestFactory, boolean clientHttpRequestFactoryIsAlreadyWrapped) {
```

Comment:
Alternatively you could always wrap and change the wrapping code to something like
```
ClientHttpRequestFactory wrapIfNeeded(ClientHttpRequestFactory factory) {
  if (factory instanceof AuthorizedClientHttpRequestFactory) return factory;
  return wrapped(factory);
}
```

This way the wrapping decision would be at one place.